### PR TITLE
Rework gentable handlers to use bsn_gentable_error.

### PR DESCRIPTION
**Pull request by kenchiang**
Reviewer: @poolakiran @andi-bigswitch

### CODE REVIEW

**Comment by andi-bigswitch** [comment]
LGTM

File: modules/OFStateManager/module/src/gentable_handlers.c [hunk 1]
```diff
@@ -270,7 +270,19 @@ ind_core_bsn_gentable_entry_add_handler(
     return;
 
 error:
-    indigo_cxn_send_bsn_error(cxn_id, obj, "Gentable add failed");
+    if (rv == INDIGO_ERROR_PARAM) {
+        indigo_cxn_send_bsn_gentable_error(cxn_id, obj, table_id,
+                                           OF_BSN_GENTABLE_ERROR_PARAM,
+                                    "Gentable add failed, invalid parameter");
+    } else if (rv == INDIGO_ERROR_RESOURCE) {
```
Comment (poolakiran) [inline modules/OFStateManager/module/src/gentable_handlers.c]: Some gentables return INDIGO_ERROR_TABLE_FULL instead of INDIGO_ERROR_RESOURCE.

**Comment by poolakiran** [comment]
Thanks for taking care of this. 
ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vbs25.eng.bigswitch.com/abat/2017.01.31.1820-m.indigo.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vbs25.eng.bigswitch.com/abat/2017.01.31.1820-m.indigo.master/abat.log
/cc @rlane

**Pull request merged**